### PR TITLE
configure blueprint 

### LINKED ISSUE #432: allow label-studio to be used in a bigger flask project (by exposing a blueprint) and correct the use of app_factory
Issue-title: allow label-studio to be used in a bigger flask project (by exposing a blueprint) and correct the use of app_factory
Description:
**Is your feature request related to a problem? Please describe.**
I'd like to be able to pip install label-studio, and consume it in an existing flask app
secondly the flask app_factory pattern is implemented incorrectly as it's immediately invoked:

https://github.com/heartexlabs/label-studio/blob/db33799ed56a3c20d54c5a486834f23e3f082d30/label_studio/server.py#L60-L76

```
app = app_factory()  # this call makes the app_factory pattern redundant, it should be left to "main()" or "gunicorn" to call
```

**Describe the solution you'd like**
using `label_studio = Blueprint()` I want to replace the `@app.route` calls with `@label_studio.route`, and move the `app_factory()` call into main

**Describe alternatives you've considered**
the "downside" of the app_factory pattern is that you can't use `@app.route` you have to use `@some_blueprint.route` instead

label-studio could opt-out of the app_factory pattern completely and therefore explicitly disallow such composition, but I think that would be a shame

**Additional context**
The docstring links to this URL so it seems that this pattern is desired.

https://flask.palletsprojects.com/en/1.1.x/patterns/appfactories/#basic-factories
https://flask.palletsprojects.com/en/1.1.x/blueprints/#why-blueprints

### PR DESCRIPTION
**Pull request by graingert**
fixes #432

### CODE REVIEW

File: label_studio/blueprint.py [hunk 1]
```diff
@@ -0,0 +1,1085 @@
...
+        # check port is busy
+        if not input_args.debug and check_port_in_use('localhost', port):
+            old_port = port
+            port = int(port) + 1
+            print('\n*** WARNING! ***\n* Port ' + str(old_port) + ' is in use.\n' +
+                  '* Trying to start at ' + str(port) +
+                  '\n****************\n')
+
+        set_web_protocol(config.get('protocol', 'http://'))
+        set_full_hostname(get_web_protocol() + host.replace('0.0.0.0', 'localhost') + ':' + str(port))
```
Comment (graingert) [inline label_studio/blueprint.py:1064]: I'd like to switch this to get the config from the `current_app.label_studio`

File: label_studio/blueprint.py [hunk 2]
```diff
@@ -0,0 +1,1085 @@
...
+                    pdir=Project.get_project_dir(input_args.project_name, input_args)))
+
+    # On `start` command, launch browser if --no-browser is not specified and start label studio server
+    if input_args.command == 'start':
+        import label_studio.utils.functions
+        import label_studio.utils.auth
+        config = Project.get_config(input_args.project_name, input_args)
+
+        # set username and password
+        label_studio.utils.auth.USERNAME = input_args.username or \
```
Comment (graingert) [inline label_studio/blueprint.py:1038]: I'd like to change this to get the username and password from  `current_app.label_studio.username` and `current_app.label_studio.password`

File: label_studio/blueprint.py [hunk 3]
```diff
@@ -0,0 +1,1085 @@
...
+        return
+
+    browser_url = ls_url + '/welcome'
+    threading.Timer(2.5, lambda: webbrowser.open(browser_url)).start()
+    print('Start browser at URL: ' + browser_url)
+
+
+def main():
+    input_args = parse_input_args()
+    app = create_app(LabelStudioConfig(input_args=input_args), set_str2datetime=True)
```
Comment (graingert) [inline label_studio/blueprint.py:1008]: is there a reason that you'd not want the str2datetime filter set? it was previously only set if main ran - so I did this for a neutral refactor

File: .github/workflows/master.yaml [hunk 4]
```diff
@@ -4,8 +4,6 @@ on:
   push:
     branches: [ master ]
   pull_request:
-    branches: [ master ]
```
Comment (graingert) [inline .github/workflows/master.yaml:7]: this change means that PRs into release branches will build

File: label_studio/blueprint.py [hunk 5]
```diff
@@ -0,0 +1,1087 @@
...
+    :param label_studio_config: The LabelStudioConfig object to use.
+    """
+    app = flask.Flask(__package__, static_url_path='')
+    app.secret_key = 'A0Zrdqwf1AQWj12ajkhgFN]dddd/,?RfDWQQT'
+    app.config['SEND_FILE_MAX_AGE_DEFAULT'] = 0
+    app.config['WTF_CSRF_ENABLED'] = False
+    app.url_map.strict_slashes = False
+    if set_str2datetime:
+        app.jinja_env.filters['str2datetime'] = str2datetime
+    app.label_studio = label_studio_config or config_from_file()
```
Comment (graingert) [inline label_studio/blueprint.py:99]: I moved `input_args` from a global into `app.label_studio.input_args`

**Comment by graingert** [comment]
@makseq can you review please?

**Comment by makseq** [comment]
Yes, we will review it for 0.7.6 version.

**Comment by kasuteru** [comment]
Hi @graingert , do you have a usage example on how to use this in a bigger flask app? I would like to try it out.

**Comment by graingert** [comment]
@makseq ?

**Comment by makseq** [comment]
@graingert Sorry, could you reopen this PR to master branch again. It was closed automatically by closing 0.7.5 release branch.

**Comment by makseq** [comment]
@graingert Could you please reopen this PR to 0.8.1 branch, I've merged it there already locally, but unexpectedly deleted 0.7.5 branch and this PR is closed automatically. I can't reopen it :(

**Comment by makseq** [comment]
@graingert Anyway it's in this commit of 0.8.1 branch: 
https://github.com/heartexlabs/label-studio/pull/472/commits/4c517d1f789bd24f3fcc1fc66dffe1d0728509dd

**Pull request closed without merging**